pkg/controller/controller.go

```diff
@@ -1361,6 +1361,30 @@ func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials *broke
 		_, err = c.kubeClient.Core().Secrets(binding.Namespace).Create(secret)
 	}
 
+	if err != nil {
+		return err
+	}
+
+	// PodPresets are immutable, so if a PodPreset already exists, it should
+	// be an error.
+	if binding.Spec.AlphaPodPresetName == nil {
+		return nil
+	}
+
+	_, err := c.kubeClient.SettingsV1alpha1().PodPresets(binding.Namespace).Get(binding.Spec.AlphaPodPresetName)
+	if err == nil {
+		return fmt.Errorf(`PodPreset "%v/%v" already exists`, binding.Namespace, binding.Spec.AlphaPodPresetName)
```

Comment:
I could be missing context, so this comment may not apply.

This could be a little bit fragile if we introduce any write logic after this that could fail.

If the write logic after this ever failed, it would never be retried because we would also bail here.

Also, does this ever get retried for any other reason.

Maybe we should we may this idempotent, and only throw an error if there is a PodPreset and it is not equal to the one in the spec.